mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/MongoDBCollection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.hibernate.ogm.datastore.mongodb.options.impl.MongoDBCollectionConverter;
+import org.hibernate.ogm.options.spi.MappingOption;
+
+/**
+ * MongoDB specific definitions for a collection.
+ *
+ * @author Guillaume Smet
+ */
+@Target({ TYPE, METHOD, FIELD })
+@Retention(RUNTIME)
+@MappingOption(MongoDBCollectionConverter.class)
+public @interface MongoDBCollection {
```

Comment:
IMHO, it would be an error to not expose the features of the underlying datastore in the most useful way. It's not about an "OGM for MongoDB", it's about OGM being useful. Each datastore has specific features and that's why people use them. It would be an error to think they are interchangeable like a standard SQL store would be (and even that is not entirely true, I was very annoyed by the fact that it was not possible to define partial indexes with JPA annotations).

Typically, the point of having Neo4j is not to store my objects in it as I would do it in MongoDB, it's about being able to leverage the features of a Graph database.

IMHO, part of our job is to do all the work to expose the interesting features of each datastore to the world using a common programming model.

BTW, you didn't answer to the point that we will probably need @MongoDBCollection to expose the options of a collection. So why not also store the index options specific to MongoDB here?